src/macros.rs

```diff
@@ -318,12 +318,11 @@ macro_rules! check_metrics {
 
             $(
                 $(
-                    let value: f64 = $true_float_value;
-                    if !value.is_nan() {
-                        assert!((func_space.metrics.$metric.$func_float() - value).abs() < f64::EPSILON);
+                    assert!(if ($true_float_value as f64).is_nan() {
+                        func_space.metrics.$metric.$func_float().is_nan()
                     } else {
-                        assert!(func_space.metrics.$metric.$func_float().is_nan());
-                    }
```

Comment:
Could you keep the old structure and just change the contents of the first if branch?

Reply:
Ok, I preferred this solution to have a compact way into a single `assert`, reducing duplicated code

Reply:
I understand, but it makes the diff more complex and is unrelated to this change. You could do it in a separate commit or PR.